app/src/main/java/com/cornellappdev/uplift/ui/components/onboarding/PhotoPicker.kt

```diff
@@ -0,0 +1,131 @@
+package com.cornellappdev.uplift.ui.components.onboarding
+
+import android.net.Uri
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.PickVisualMediaRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.R
+
+/***
+ * Component that displays a camera icon for the user to select a photo for their profile
+ *  @param onPhotoSelected: function to call when the user selects a photo. Takes in uri parameter
+ */
+@Composable
+fun PhotoPicker(onPhotoSelected: (Uri) -> Unit){
+    // State to store the selected image URI
+    var selectedImageUri by rememberSaveable { mutableStateOf<Uri?>(null) }
+
+    // Registers a photo picker activity launcher in single-select mode.
+    val pickMedia = rememberLauncherForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+        // Callback is invoked after the user selects a media item or closes the
+        // photo picker.
+        if (uri != null) {
+            selectedImageUri = uri
+            onPhotoSelected(uri)
+        }
+    }
+
+    Box(
+        modifier = Modifier
+            .size(160.dp),
+        contentAlignment = Alignment.Center
+    ){
```

Comment:
I sniff unformatted file >:)

You can run Ctrl+Alt+L or equivalent on mac to reformat the file. Or, right click on the file and hit reformat code.

Reply:
Whoops, didn't know this was a thing, cool feature, will be sure to use in the future a lot more lol